### PR TITLE
CM-40909 - Add lock file restore for Golang, NPM and Nuget

### PR DESCRIPTION
**Pull request by naftalicy**
…mmands execution

### CODE REVIEW

File: cycode/cli/files_collector/sca/base_restore_dependencies.py [hunk 1]
```diff
@@ -85,7 +92,7 @@ def is_project(self, document: Document) -> bool:
         pass
 
     @abstractmethod
-    def get_command(self, manifest_file_path: str) -> List[str]:
+    def get_commands(self, manifest_file_path: str) -> List[str]:
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/base_restore_dependencies.py:95]: `List[List[str]]` instead?

File: cycode/cli/files_collector/sca/base_restore_dependencies.py [hunk 2]
```diff
@@ -13,19 +13,26 @@ def build_dep_tree_path(path: str, generated_file_name: str) -> str:
...
+            dependencies = shell(command=command, timeout=command_timeout, working_directory=working_directory)
+            all_dependencies.append(dependencies)  # Collect each command's output
+
+        # Write all collected outputs to the file if dependencies_file_name is provided
         if dependencies_file_name:
-            with open(dependencies_file_name, 'w') as output_file:
-                output_file.write(dependencies)
+            with open(dependencies_file_name, 'w') as output_file:  # Open once in 'w' mode to start fresh
+                for dependencies in all_dependencies:
+                    output_file.write(dependencies + '\n')
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/base_restore_dependencies.py:35]: it is more clean to use `writelines` instead of `write`

**Review by MarshalX — COMMENTED** [review]
well done

File: cycode/cli/files_collector/sca/base_restore_dependencies.py (hunk 2 above)
Comment (naftalicy) [reply cycode/cli/files_collector/sca/base_restore_dependencies.py]: done

File: cycode/cli/files_collector/sca/base_restore_dependencies.py [hunk 3]
```diff
@@ -13,19 +13,26 @@ def build_dep_tree_path(path: str, generated_file_name: str) -> str:
...
+            dependencies = shell(command=command, timeout=command_timeout, working_directory=working_directory)
+            all_dependencies.append(dependencies)  # Collect each command's output
+
+        # Write all collected outputs to the file if dependencies_file_name is provided
         if dependencies_file_name:
-            with open(dependencies_file_name, 'w') as output_file:
-                output_file.write(dependencies)
+            with open(dependencies_file_name, 'w') as output_file:  # Open once in 'w' mode to start fresh
+                for dependencies in all_dependencies:
+                    output_file.writelines(dependencies)
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/base_restore_dependencies.py:35]: Am I wrong about it?

```suggestion
                output_file.writelines(all_dependencies)
```

Comment (naftalicy) [reply cycode/cli/files_collector/sca/base_restore_dependencies.py]: After I checked again we do need a new line \n for the multi command generation, I move it upward inside the append code, then wI changed the actual file writing to `output_file.writelines(all_dependencies)` without for loop

**Review by MarshalX — APPROVED** [review]

**Pull request merged**